### PR TITLE
feat(filter): domain-year-mail

### LINKED ISSUE #29: Visualiser la liste des expéditeurs et des destinataires
### EPIC

https://github.com/SocialGouv/archifiltre-private/issues/3

### AFTER

#28 

### Contexte

Dans le contexte de la visualisation du contenu de la messagerie qui permet de donner du sens pour la compréhension du contenu

### Définition du(des) problème(s)

Face à une liste des éléments extraits du pst, il est impossible de savoir qui sont les acteurs importants en lien avec le propriétaire de la messagerie

### Solution

ETQ utilisateur Archimail
JVP visualiser les expéditeurs des messages reçus et les destinataires des messages envoyés en visualisant sous forme de Bubbles/circle Packing (dégradé de bleu):
- par grands domaines mails (sante.gouv/beta.gouv/gmail, etc.)
- par nom des utilisateurs (expéditeurs et/ou destinataires)
- par années
AD pouvoir comprendre les orientations du contenu du pst

### Critères d'acceptance

_No response_

### Les tests e2e doivent-ils passer ?

- [ ] e2e

### Valeur Business

BV:M

### LINKED ISSUE #78: Compléter le bloc "Informations" de l'interface
### EPIC

https://github.com/SocialGouv/archifiltre-private/issues/3

### AFTER

https://github.com/SocialGouv/archimail/issues/45

### Contexte

Quand un utilisateur dépose un pst, il peut visualiser son contenu dans l'interface. S'il clique sur un élément (mail ou dossier), il peut actuellement afficher:
- Nom du dossier/mail
- Nombre de PJ
- Nombre de mails
- Date d'envoi
- Date de réception
- Adresse mail

### Définition du(des) problème(s)

Dans cet affichage des informations de l'élément, il manque des informations prévues dans la maquette et d'autres informations.

### Solution

ETQ utilisateur d'Archifiltre-Mails
JVP visualiser davantage de métadonnées dans la partie "Informations"
- Nature de l'élément : dossier ou mail
- Titre de l'élément
- Adresse mail de l'expéditeur
- Adresses mails du ou des destinataires directs et en copie
- Date d'envoi
- Date de réception
A droite de l'affichage:
- Nombre de mails
- Nombre de pièces jointes
- Volume occupé et pourcentage
AD mieux appréhender le contenu et identifier les contenus à éliminer

### Critères d'acceptance

_No response_

### Les tests e2e doivent-ils passer ?

- [ ] e2e

### Valeur Business

BV:L

### LINKED ISSUE #82: Afficher le fil d'Ariane en entier
### EPIC

https://github.com/SocialGouv/archifiltre-private/issues/3

### AFTER

https://github.com/SocialGouv/archimail/issues/45

### Contexte

Quand on dépose un pst, on affiche sous forme de Bubbles l'ensemble des mails par domaine puis par correspondant puis par année puis l'ensemble des mails concerné.

### Définition du(des) problème(s)

Dans la maquette, un fil d'Ariane reprenant les éléments affichés (Domaine/Correspondant/Année/Elément) est prévu.

### Solution

ETQ utilisateur d'Archifiltre-Mails
JVP visualiser dans quels niveaux je me situe en visualisant le fil d'Ariane en haut qui reprend visuellement tous les éléments affichés sous la forme
domaine > correspondant > année > élément
AD pouvoir comprendre ce que je visualise

### Critères d'acceptance

_No response_

### Les tests e2e doivent-ils passer ?

- [ ] e2e

### Valeur Business

BV:L

### PR DESCRIPTION
**Pull request by mehdilouraoui**
Closes #29 
Closes #78 
Closes #82

# Objectif(s)

### CODE REVIEW

**Comment by lsagetlethias** [comment]
Please consider resolving comments from https://github.com/SocialGouv/archimail/pull/65 in this current PR

**Comment by lsagetlethias** [comment]
https://github.com/SocialGouv/archifiltre-mails/releases/tag/v0.0.1-feat-filter-domain-year-mail.0

**Comment by lsagetlethias** [comment]
⬆️ To upgrade:
- jotai
- @nivo/generators
- d3
- react-dropzone

File: src/renderer/components/common/card/Card.module.scss [hunk 1]
```diff
@@ -9,8 +9,8 @@ $dashboard-card-padding: 2rem 3rem;
   padding: $dashboard-card-padding;
   border-radius: 10px;
   background: $background-color-white;
-  min-width: 625px;
-  position: relative;
+  position: absolute;
+  font-size: 12px;
```
Comment (lsagetlethias) [inline src/renderer/components/common/card/Card.module.scss:13]: Shouldn't be constants?

File: src/renderer/components/common/card/Card.tsx [hunk 2]
```diff
@@ -5,12 +5,21 @@ import style from "./Card.module.scss";
...
+    className?: string;
     color?: "blue" | "green" | "grey" | "orange" | "purple";
 }
 
-export const Card: FC<CardProps> = ({ children, title, color = "grey" }) => {
+export const Card: FC<CardProps> = ({
+    children,
+    title,
+    className,
+    color = "grey",
```
Comment (lsagetlethias) [inline src/renderer/components/common/card/Card.tsx:16]: Magic color (design system)

File: src/renderer/components/common/card/Card.tsx [hunk 3]
```diff
@@ -5,12 +5,21 @@ import style from "./Card.module.scss";
...
+}) => {
+    const _className = className
```
Comment (lsagetlethias) [inline src/renderer/components/common/card/Card.tsx:18]:
```suggestion
    const cardClassName = className
```

File: src/renderer/components/menu/Menu.tsx [hunk 4]
```diff
@@ -1,21 +1,34 @@
...
             >
-                <li id="to-delete-btn">Supprimé</li>
-                <li>Non marqué</li>
-                <li id="to-keep-btn">Conservé</li>
+                <li
+                    onClick={addMarkedToDelete}
+                    id="to-delete-btn"
+                    className="to-delete-btn"
+                >
+                    Supprimé
```
Comment (lsagetlethias) [inline src/renderer/components/menu/Menu.tsx:26]: Add TODO for i18n

File: src/renderer/components/menu/Menu.tsx [hunk 5]
```diff
@@ -1,21 +1,34 @@
...
+                </li>
+                {/* <li>Non marqué</li> */}
```
Comment (lsagetlethias) [inline src/renderer/components/menu/Menu.tsx:28]: Delete

File: src/renderer/components/viewer/CirclePacking.tsx [hunk 6]
```diff
@@ -1,25 +1,128 @@
...
+                ...getChildrenToKeepIds(children, markedToDelete),
+                ...getUntagChildrenIds(children, markedToKeep, markedToDelete),
+            ]);
+        }
+    }, [currentView]); // TODO: Investigate
+
+    const getTaggedFilesColor = (
+        node: Omit<ComputedDatum<ViewerObject<string>>, "color" | "fill">
+    ) => {
+        if (node.depth === 0) return BASE_COLOR_LIGHT; // prefer to use equality over a '!node.depth' to understand that is a level
```
Comment (lsagetlethias) [inline src/renderer/components/viewer/CirclePacking.tsx:71]: Delete comment

Comment (lsagetlethias) [reply src/renderer/components/viewer/CirclePacking.tsx:71]: or not, you choose 😅

File: src/renderer/components/viewer/CirclePackingTooltip.tsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import type { ComputedDatum } from "@nivo/circle-packing";
+import React from "react";
+import type { DefaultViewerObject } from "src/renderer/utils/pst-extractor";
```
Comment (lsagetlethias) [inline src/renderer/components/viewer/CirclePackingTooltip.tsx:3]: Change import

File: src/renderer/hooks/useDomainsYearMails.tsx [hunk 8]
```diff
@@ -0,0 +1,137 @@
+import { useCallback, useEffect, useState } from "react";
+
+import { usePstStore } from "../store/PSTStore";
+import { CORRESPONDANTS, DOMAIN, MAILS, YEAR } from "../utils/constants";
```
Comment (lsagetlethias) [inline src/renderer/hooks/useDomainsYearMails.tsx:4]: Separate consts

File: src/renderer/hooks/useDomainsYearMails.tsx [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+                    +node.data.name
+                );
+
+            setCurrentView({
+                elements: createMails(mailsByYearAndCorrespondant, node.id),
+                type: MAILS,
+            });
+        }
+
+        return void 0;
```
Comment (lsagetlethias) [inline src/renderer/hooks/useDomainsYearMails.tsx:129]:
```suggestion
        return;
```

File: src/renderer/store/PSTStore.tsx [hunk 10]
```diff
@@ -1,82 +1,65 @@
...
     extractTables: PstExtractTables | undefined;
     setExtractTables: (
         update: SetStateAction<PstExtractTables | undefined>
     ) => void;
-    mainInfos: PstEmail | undefined;
-    setMainInfos: (update: SetStateAction<PstEmail | undefined>) => void;
-    depth: number | string;
-    setDepth: (update: SetStateAction<number>) => void;
+    mainInfos: MainInfos | undefined;
+    setMainInfos: (update?: SetStateAction<MainInfos | undefined>) => void;
```
Comment (lsagetlethias) [inline src/renderer/store/PSTStore.tsx:33]: It feels weird to have `update` possibly undefined, or other `set` methods

File: src/renderer/utils/pst-viewer.ts [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+            ? `${node.data.email.receivedDate?.getDay()}-${node.data.email.receivedDate?.getMonth()}-${node.data.email.receivedDate?.getFullYear()}`
+            : node.data.name,
+    labelsFilter: (label) => label.node.height === 0,
+    labelsSkipRadius: 16,
+    motionConfig: "slow",
+    padding: 2,
+    value: "size",
+};
+
+export const sanitizeMailDate = (date: Date): string =>
```
Comment (lsagetlethias) [inline src/renderer/utils/pst-viewer.ts:90]: 📅 use [Date.prototype.toLocalDateString()](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString) or event better, [Intl.DateTimeFormat()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat) in combinaison with i18n module

File: src/renderer/views/dashboard/DashboardViewer.tsx [hunk 12]
```diff
@@ -1,15 +1,20 @@
...
-            <DashboardViewerBreadcrumb />
-            <DashboardViewerCircle />
-        </Card>
-    </div>
+    <Card title="Visualisation" color="grey">
+        <DashboardViewerBreadcrumb />
+        <DashboardViewerCircle />
+    </Card>
 );
+// export const DashboardViewer: React.FC = () => (
```
Comment (lsagetlethias) [inline src/renderer/views/dashboard/DashboardViewer.tsx:13]: Delete?

**Review by lsagetlethias — CHANGES_REQUESTED** [review]
Great job overall 👍 

Don't forget to handle the most `TODO` you can! 😜

File: src/renderer/components/common/card/Card.module.scss (hunk 1 above)
Comment (mehdilouraoui) [reply src/renderer/components/common/card/Card.module.scss:13]: done

File: src/renderer/components/common/card/Card.tsx (hunk 3 above)
Comment (mehdilouraoui) [reply src/renderer/components/common/card/Card.tsx:18]: done

File: src/renderer/components/common/card/Card.tsx (hunk 2 above)
Comment (mehdilouraoui) [reply src/renderer/components/common/card/Card.tsx:16]: done (design system to implement later tho)

File: src/renderer/components/menu/Menu.tsx (hunk 4 above)
Comment (mehdilouraoui) [reply src/renderer/components/menu/Menu.tsx:26]: done

File: src/renderer/components/menu/Menu.tsx (hunk 5 above)
Comment (mehdilouraoui) [reply src/renderer/components/menu/Menu.tsx:28]: done

File: src/renderer/components/viewer/CirclePacking.tsx (hunk 6 above)
Comment (mehdilouraoui) [reply src/renderer/components/viewer/CirclePacking.tsx:71]: erased 😅

File: src/renderer/components/viewer/CirclePackingTooltip.tsx (hunk 7 above)
Comment (mehdilouraoui) [reply src/renderer/components/viewer/CirclePackingTooltip.tsx:3]: done

File: src/renderer/hooks/useDomainsYearMails.tsx (hunk 9 above)
Comment (mehdilouraoui) [reply src/renderer/hooks/useDomainsYearMails.tsx:129]: done

File: src/renderer/store/PSTStore.tsx (hunk 10 above)
Comment (mehdilouraoui) [reply src/renderer/store/PSTStore.tsx:33]: jotai acts weird on useState... need to investigate on this but `PstStore` will be completely refactor asap!

File: src/renderer/utils/pst-viewer.ts (hunk 11 above)
Comment (mehdilouraoui) [reply src/renderer/utils/pst-viewer.ts:90]: Yes, this sanitize function is a bit awkward and "under-engineer"... 😅 but yes, should be modify with i18n module when translation will be under construction!

**Review by lsagetlethias — APPROVED** [review]

**Pull request merged**